bot/exts/summer_aoc.py

```diff
@@ -0,0 +1,323 @@
+        self.post_time = 0
+
+        self.bot.loop.create_task(self.load_event_state())
+
+    def is_configured(self) -> bool:
+        """Check whether all the necessary settings are configured to run the event."""
+        return None not in (self.year, self.current_day, self.day_interval, self.post_time)
+
+    async def cog_check(self, ctx: commands.Context) -> bool:
+        """Role-lock all commands in this cog."""
+        return await commands.has_any_role(
+            Roles.admins,
+            Roles.events_lead,
+            Roles.event_runner,
+        ).predicate(ctx)
+
+    async def cog_command_error(self, ctx: commands.Context, error: commands.CommandError) -> None:
+        """Send help text on user input error."""
+        if isinstance(error, commands.UserInputError):
+            desc = f"```{Client.prefix}summeraoc {ctx.command.name} {ctx.command.signature}```"
+            embed = discord.Embed(
+                description=desc,
+            )
+            await ctx.send(embed=embed)
+
+    @commands.group(invoke_without_command=True, name="summeraoc", aliases=["roc", "revivalofcode"])
+    async def summer_aoc_group(self, ctx: commands.Context) -> None:
+        """Commands for managing the Summer AoC event."""
+        desc = "\n".join(
+            f"*{command.help}*\n```{Client.prefix}summeraoc {command.name} {command.signature}```"
+            for command in sorted(self.summer_aoc_group.walk_commands(), key=hash)
+        )
+        embed = discord.Embed(description=desc)
+        await ctx.send(embed=embed)
+
+    @summer_aoc_group.command(name="info")
+    async def info(self, ctx: commands.Context) -> None:
+        """Display info about the state of the event."""
+        embed = self.get_info_embed()
+        await ctx.send(embed=embed)
+
+    @summer_aoc_group.command(name="start")
+    async def start(self, ctx: commands.Context, year: int, day_interval: int, post_time: int = 0) -> None:
+        """
+        Start the Summer AoC event.
+        To specify a starting day other than `1`, use the `force` command.
+        
+        `year` must be an integer at least 2015.
+        `day_interval` must be an integer at least 1.
+        `post_time` must be an integer between 0 and 23.
+        """  # noqa: D205
+        if not FIRST_YEAR <= year <= LAST_YEAR:
+            raise commands.BadArgument(f"Year must be between {FIRST_YEAR} and {LAST_YEAR}, inclusive")
+
+        if day_interval < 1:
+            raise commands.BadArgument("Day interval must be at least 1")
+
+        if not 0 <= post_time <= 23:
+            raise commands.BadArgument("Post time must be between 0 and 23")
+
+        if self.is_running:
+            await ctx.send("A Summer AoC event is already running!")
+            return
+
+        self.is_running = True
+        self.year = year
+        self.current_day = 1
+        self.day_interval = day_interval
+        self.post_time = post_time
```

Comment:
We can't change the `post_time` once this starts. Is this a feature we want to implement in case we need to adjust the posting time on the fly?

Reply:
My thinking is that we can use the force command either as a one-time thing or by stopping the event and starting a new one with a different post-time.